discovery-infra/test_infra/controllers/node_controllers/node.py

```diff
@@ -49,9 +51,26 @@ def macs(self):
 
     @property
     def ssh_connection(self):
-        return ssh.SshConnection(self.ips[0],
-                                 private_ssh_key_path=self.private_ssh_key_path,
-                                 username=self.username)
+        if not self.ips:
+            raise RuntimeError(f"No available IPs for node {self.name}")
+
+        logging.info("Trying to access through IP addresses: %s", ", ".join(self.ips))
+        for ip in self.ips:
+            exception = None
+            try:
+                connection = ssh.SshConnection(
+                    ip,
+                    private_ssh_key_path=self.private_ssh_key_path,
+                    username=self.username)
+                connection.connect()
+                return connection
+
+            except (TimeoutError, SCPException) as e:
```

Comment:
sure you don't want to catch generic exception too? I think much better to have logs than to right code nicer

Reply:
See the CI channel, we always have the logs coming after sosreport step
If we'll see it's getting on our way we should consider changing that